src/bundles/peer-bandwidth.js

```diff
@@ -0,0 +1,172 @@
+    },
+
+    selectPeerBandwidthNow: state => state.peerBandwidth.now,
+    selectPeerBandwidthUpdatingPeerIds: state => state.peerBandwidth.updatingPeerIds,
+    selectPeerBandwidthPeers: state => state.peerBandwidth.peers,
+
+    // Select the next peer to update from the list of peers we know about, only
+    // if we're not already updating the maximum number of peers we're allowed to
+    // update at once, and only if there is a peer that is ready to be updated
+    // (enough time has elapsed since last update).
+    selectNextPeerIdForUpdate: createSelector(
+      'selectPeerBandwidthNow',
+      'selectPeerBandwidthPeers',
+      'selectPeerBandwidthUpdatingPeerIds',
+      (now, peers, updatingPeerIds) => {
+        if (!peers.length || updatingPeerIds.length >= opts.peerUpdateConcurrency) {
+          return null
+        }
+
+        const lastAttempt = p => Math.max(p.lastSuccess || 0, p.lastFailure || 0)
+        const isInactive = p => p.bw ? p.bw.rateIn.eq(0) && p.bw.rateOut.eq(0) : false
+        const peerUpdateInterval = p => isInactive(p)
+          ? opts.inactivePeerUpdateInterval
+          : opts.peerUpdateInterval
+        const isTimeToUpdate = p => lastAttempt(p) + peerUpdateInterval(p) < now
+        const isNotAlreadyUpdating = p => !updatingPeerIds.includes(p.id)
+
+        const peer = peers
+          .filter(isTimeToUpdate)
+          .sort((a, b) => lastAttempt(b) - lastAttempt(a))
+          .filter(isNotAlreadyUpdating)[0]
+
+        return peer ? peer.id : null
+      }
+    ),
+
+    // Get updated bandwidth for a particular peer
+    doUpdatePeerBandwidth: peerId => async ({ dispatch, getState, getIpfs }) => {
+      dispatch({ type: 'UPDATE_PEER_BANDWIDTH_STARTED', payload: { peerId } })
+
+      const ipfs = getIpfs()
+      let bw
+
+      try {
+        bw = await ipfs.stats.bw({ peer: peerId })
+      } catch (err) {
+        return dispatch({
+          type: 'UPDATE_PEER_BANDWIDTH_FAILED',
+          payload: { peerId, error: err }
+        })
+      }
+
+      dispatch({ type: 'UPDATE_PEER_BANDWIDTH_FINISHED', payload: { peerId, bw } })
+    },
+
+    reactUpdatePeerBandwidth: createSelector(
+      'selectIpfsReady',
+      'selectNextPeerIdForUpdate',
+      (ipfsReady, peerId) => {
+        if (ipfsReady && peerId) {
+          return { actionCreator: 'doUpdatePeerBandwidth', args: [peerId] }
+        }
+      }
+    ),
+
+    reactSyncPeers: createSelector(
+      'selectPeers',
+      'selectPeerBandwidthPeers',
+      (peers, bwPeers) => {
+        const peerIds = (peers || []).reduce((ids, p) => {
+          const id = p.peer.toB58String()
+          if (ids.seen[id]) return ids
+          ids.seen[id] = true
+          ids.unique.push(id)
+          return ids
+        }, { seen: {}, unique: [] }).unique
+
+        const added = peerIds.filter(id => !bwPeers.some(p => p.id === id))
+        const removed = bwPeers.filter(p => peerIds.every(id => id !== p.id)).map(p => p.id)
```

Comment:
You can make `removed` more like `added` and get the stop early benefits of `some` again.
```js
const removed = bwPeers.filter(p => !peerIds.some(id => id === p.id)).map(p => p.id)
```